File: Terraria/decrypt.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad
import os


def decrypt_file(input_file, output_file):
    try:
        # Equivalent to C# UnicodeEncoding (UTF-16LE)
        key_string = "h3y_gUyZ"
        key = key_string.encode("utf-16le")

        # RijndaelManaged defaults:
        # BlockSize = 128
        # Mode = CBC
        # Padding = PKCS7
        cipher = AES.new(key, AES.MODE_CBC, key)

        with open(input_file, "rb") as f:
            encrypted_data = f.read()

        decrypted_data = cipher.decrypt(encrypted_data)

        # Remove PKCS7 padding (same as .NET default)
        decrypted_data = unpad(decrypted_data, AES.block_size)

        with open(output_file, "wb") as f:
            f.write(decrypted_data)

        return False  # matches C# return on success

    except Exception:
        if os.path.exists(output_file):
            os.remove(output_file)
        return True  # matches C# return on failure


if __name__ == "__main__":
    print(decrypt_file("Calli.plr", "decrypt.plr"))

File: Terraria/decrypt.java
import javax.crypto.Cipher;
import javax.crypto.CipherInputStream;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.io.*;
import java.nio.charset.StandardCharsets;
import java.nio.charset.Charset;

public class DecryptFile {

    public static boolean decryptFile(String inputFile, String outputFile) {
        try {
            // Equivalent to C# UnicodeEncoding (UTF-16LE)
            String keyString = "h3y_gUyZ";
            byte[] keyBytes = keyString.getBytes(Charset.forName("UTF-16LE"));

            SecretKeySpec keySpec = new SecretKeySpec(keyBytes, "AES");
            IvParameterSpec ivSpec = new IvParameterSpec(keyBytes);

            // AES/CBC/PKCS5Padding in Java = PKCS7 compatible
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            cipher.init(Cipher.DECRYPT_MODE, keySpec, ivSpec);

            FileInputStream fis = new FileInputStream(inputFile);
            CipherInputStream cis = new CipherInputStream(fis, cipher);
            FileOutputStream fos = new FileOutputStream(outputFile);

            int b;
            while ((b = cis.read()) != -1) {
                fos.write(b);
            }

            fos.close();
            cis.close();
            fis.close();

            return false; // success (matches C# behavior)

        } catch (Exception e) {
            new File(outputFile).delete();
            return true; // failure (matches C# behavior)
        }
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.out.println("Usage: java DecryptFile inputFile outputFile");
            return;
        }

        boolean failed = decryptFile(args[0], args[1]);
        if (failed) {
            System.out.println("Decryption failed.");
        } else {
            System.out.println("Decryption successful.");
        }
    }
}
